frontend/src/context/CartContext.test.tsx

```diff
@@ -0,0 +1,250 @@
+import { act, renderHook, type RenderHookResult } from "@testing-library/react";
+import { expect, vi } from "vitest";
+
+import { CartProvider, useCart } from "./CartContext";
+import { getLocalData, setLocalData } from "./utils/localStorage";
+
+vi.mock("./utils/localStorage", () => ({
+  getLocalData: vi.fn(),
+  setLocalData: vi.fn(),
+}));
+
+/**
+ * Helper function to render the CartProvider with a specific initial cart state
+ * @param initialCartValue - The initial cart value to use in the context
+ * @default initialCartValue { nw: {}, pns: {}, wls: {} }
+ */
+function renderComponentWithCart(
+  initialCartValue: Cart | null = { nw: {}, pns: {}, wls: {} },
+): RenderHookResult<ReturnType<typeof useCart>, unknown> {
+  vi.mocked(getLocalData).mockReturnValue(initialCartValue);
+
+  const wrapper = ({ children }: { children: React.ReactNode }) => (
+    <CartProvider>{children}</CartProvider>
+  );
+
+  return renderHook(() => useCart(), { wrapper });
+}
+
+const fullProductDetails = {
+  id: "fake_id",
+  title: "Product Name 500g",
+  image: "https://example.com/product/p1.jpg",
+  productPageUrl: "https://example.com/product/p1",
+  price: {
+    value: "4.29",
+    per: "ea",
+    unitPrice: "0.86",
+    unit: "100g",
+  },
+  promo: null,
+} as Product;
+
+const existingCart = {
+  nw: {
+    nw1: {
+      product: fullProductDetails,
+      quantity: 5,
+    },
+  },
+  pns: {
+    pns1: {
+      product: fullProductDetails,
+      quantity: 12,
+    },
+    pns2: {
+      product: fullProductDetails,
+      quantity: 1,
+    },
+  },
+  wls: {},
+};
+
+// ----- ACTIVE INTERACTIONS -----
```

Comment:
Once other things have moved into their own files this comment can be removed :)